crates/settlement-clients/ethereum/src/tests/mod.rs

```diff
@@ -0,0 +1,318 @@
+    assert_eq!(input_bytes, expected);
+}
+
+// SOLIDITY FUNCTIONS NEEDED
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    STARKNET_CORE_CONTRACT,
+    "src/test_data/ABI/starknet_core_contract.json"
+);
+
+sol! {
+    #[allow(missing_docs)]
+    #[sol(rpc, bytecode="6080604052348015600e575f80fd5b506101c18061001c5f395ff3fe608060405234801561000f575f80fd5b5060043610610029575f3560e01c8063b72d42a11461002d575b5f80fd5b6100476004803603810190610042919061010d565b610049565b005b50505050565b5f80fd5b5f80fd5b5f80fd5b5f80fd5b5f80fd5b5f8083601f84011261007857610077610057565b5b8235905067ffffffffffffffff8111156100955761009461005b565b5b6020830191508360208202830111156100b1576100b061005f565b5b9250929050565b5f8083601f8401126100cd576100cc610057565b5b8235905067ffffffffffffffff8111156100ea576100e961005b565b5b6020830191508360018202830111156101065761010561005f565b5b9250929050565b5f805f80604085870312156101255761012461004f565b5b5f85013567ffffffffffffffff81111561014257610141610053565b5b61014e87828801610063565b9450945050602085013567ffffffffffffffff81111561017157610170610053565b5b61017d878288016100b8565b92509250509295919450925056fea2646970667358221220fa7488d5a2a9e6c21e6f46145a831b0f04fdebab83868dc2b996c17f8cba4d8064736f6c634300081a0033")]
+    contract DummyCoreContract {
+        function updateStateKzgDA(uint256[] calldata programOutput, bytes calldata kzgProof)  external {
+        }
+    }
+}
+
+// UTILITY FUNCTIONS NEEDED
+
+fn get_program_output(block_no: u64) -> Vec<[u8; 32]> {
+    // Program Output
+    let program_output_file_path =
+        format!("{}{}{}{}", *CURRENT_PATH, "/src/test_data/program_output/", block_no, ".txt");
+
+    let mut program_output: Vec<[u8; 32]> = Vec::new();
+    let file = File::open(program_output_file_path).expect("Failed to read program output file");
+    let reader = BufReader::new(file);
+
+    for line in reader.lines() {
+        let line = line.expect("can't read line");
+        let trimmed = line.trim();
+        assert!(!trimmed.is_empty());
+
+        let result: [u8; 32] = U256::from_str(trimmed)
+            .expect("Unable to convert line")
+            .to_be_bytes_vec()
+            .as_slice()
+            .pipe(to_padded_hex)
+            .pipe(|hex| hex_string_to_u8_vec(&hex).expect("unable to convert"))
+            .try_into()
+            .expect("Vector length must be 32");
+
+        program_output.push(result)
+    }
+    program_output
+}
+
+fn get_blob_data(block_no: u64) -> Vec<Vec<u8>> {
+    // Blob Data
+    let blob_data_file_path = format!("{}{}{}{}", *CURRENT_PATH, "/src/test_data/blob_data/", block_no, ".txt");
+    let blob_data = fs::read_to_string(blob_data_file_path).expect("Failed to read the blob data txt file");
+    let blob_data_vec = vec![hex_string_to_u8_vec(&blob_data).unwrap()];
+    blob_data_vec
+}
+
+fn hex_string_to_u8_vec(hex_str: &str) -> color_eyre::Result<Vec<u8>> {
```

Comment:
it feels like we have a lot of conversion functions now and we might

1. have some duplicates
2. have ways to reduce code using generics (instead of implement a method for both size 32 and size 48, we can use a generic maybe)

not for this PR, maybe we can create an `enhancement` issue

Reply:
Noted. 
you can find the issue here:  #84